### PR TITLE
Add Google Cloud ML Diagnostics metrics support and documentation

### PR DESCRIPTION
**Pull request by richaguptaa17**
Integrate Google Cloud ML Diagnostics SDK (google-cloud-mldiagnostics) into MaxDiffusion to automatically record training, system, and performance metrics.

Key Changes:
- train_utils.py: Added _METRICS_TO_MANAGED mapping table converting metrics names to canonical MetricType enums (loss, learning_rate, gradient_norm, total_weights, step_time, tflops) with automatic pass-through for custom metrics. 
- max_utils.py: configured region=None for GCP cluster auto-discovery, and enabled background hardware metric collection (log_system_metrics=True).
- metrics.md: Created comprehensive guide for capturing metrics using google-cloud-mldiagnostics integration in training scripts.

Tested:
- Ran a multi-host distributed training run on TPU v6e cluster. Verified successful metric ingestion in Cloud Logging for predefined, custom, and hardware utilization metrics.

### CODE REVIEW

**Review by entrpn — DISMISSED** [review]

File: src/maxdiffusion/max_utils.py [hunk 1]
```diff
@@ -88,28 +88,47 @@ def _jax_profiler_enabled(config):
   return "enable_profiler" in config.get_keys() and config.enable_profiler and jax.process_index() == 0
 
 
-def _ml_diagnostics_profiler_enabled(config):
+def ml_diagnostics_enabled(config):
   return "enable_ml_diagnostics" in config.get_keys() and config.enable_ml_diagnostics
 
 
+_ml_diagnostics_profiler_enabled = ml_diagnostics_enabled
```
Comment (mbohlool) [inline src/maxdiffusion/max_utils.py]: Why do you need this?

File: src/maxdiffusion/train_utils.py [hunk 2]
```diff
@@ -76,12 +76,32 @@ def _validate_gcs_bucket_name(bucket_name, config_var):
...
 
-def record_scalar_metrics(metrics, step_time_delta, per_device_tflops, lr):
+try:
+  from google_cloud_mldiagnostics import metrics as mld_metrics, metric_types
+except ImportError:
+  mld_metrics = None
+  metric_types = None
+
+
+_metric_type_enum = getattr(metric_types, "MetricType", None) if metric_types else None
```
Comment (mbohlool) [inline src/maxdiffusion/train_utils.py]: do we expect MetricType attribute not present if metric_types is not None? Lots of silence failure here. If ml diagnostic exists, we should not ignore any failure after that. So if I install ml_diagnostic dependencies on a job, I expect it to work and if there is any incompatibilities/failures, the job should fail rather than ignore it and disable ml_diagnostics.

**Comment by Perseus14** [comment]
Thanks for the PR @richaguptaa17. Could you please do the following?

- Run linting commands `ruff check .` and `pyink src/maxdiffusion --pyink-indentation=2 --line-length=125` (ruff==0.15.12 and pyink==23.10.0)
- Squash commits

File: src/maxdiffusion/max_utils.py (hunk 1 above)
Comment (richaguptaa17) [reply src/maxdiffusion/max_utils.py]: Removed. It was an unnecessary alias from renaming the function.

File: src/maxdiffusion/train_utils.py (hunk 2 above)
Comment (richaguptaa17) [reply src/maxdiffusion/train_utils.py]: Agreed. If google_cloud_mldiagnostics is installed, metric_types.MetricType should be present. Replaced fallback with direct mapping.

Made a change to raise an explicit ImportError if a user sets `enable_ml_diagnostics: True` but google_cloud_mldiagnostics is not installed or initialized.

**Comment by richaguptaa17** [comment]
Ran the linting commands and squashed the commits.

File: src/maxdiffusion/max_utils.py (hunk 1 above)
Comment (mbohlool) [reply src/maxdiffusion/max_utils.py]: I can still see it, maybe you forgot to push?

**Review by entrpn — DISMISSED** [review]

File: src/maxdiffusion/max_utils.py (hunk 1 above)
Comment (richaguptaa17) [reply src/maxdiffusion/max_utils.py]: Done, verified.

**Review by Perseus14 — DISMISSED** [review]

**Review by mbohlool — APPROVED** [review]

**Review by entrpn — APPROVED** [review]

File: src/maxdiffusion/__init__.py [hunk 3]
```diff
@@ -47,8 +47,6 @@
     "configuration_utils": ["ConfigMixin"],
     "max_logging": [],
     "max_utils": [],
-    "maxdiffusion_google": [],
```
Comment (mbohlool) [inline src/maxdiffusion/__init__.py:50]: why you removed these two lines?

**Review by mbohlool — CHANGES_REQUESTED** [review]

**Review by mbohlool — APPROVED** [review]

File: src/maxdiffusion/__init__.py (hunk 3 above)
Comment (richaguptaa17) [reply src/maxdiffusion/__init__.py:50]: These are Google3-internal modules that do not exist in the open-source GitHub repository.
